js/modules/loki_message_api.js

```diff
@@ -164,6 +186,12 @@ class LokiMessageAPI {
           await lokiSnodeAPI.updateSwarmNodes(params.pubKey, newSwarm);
           this.sendingSwarmNodes[params.timestamp] = newSwarm;
           return false;
+        } else if (e instanceof textsecure.WrongDifficultyError) {
+          const { newDifficulty } = e;
+          if (!Number.isNaN(newDifficulty)) {
+            window.storage.put('PoWDifficulty', newDifficulty);
+          }
+          throw e;
```

Comment:
do you want to throw even after updating the difficulty?
Otherwise `return false;` in the `if` block?

Reply:
Ya I explicitly throw here to force a restart of the sendMessage logic to actually *use* the new PoW difficulty
```js
    try {
      results = await Promise.all(promises);
    } catch (e) {
      if (e instanceof textsecure.WrongDifficultyError) {
        // Force nonce recalculation
        this.sendMessage(pubKey, data, messageTimeStamp, ttl, options);
        return;
      }
      throw e;
    }
```
Not the most elegant solution but gets the job done, can also be extended to different errors in the future to do different things if we want

Reply:
Seems reasonable!